### DISCUSSION TITLE
Debug on VS Code for IBMi

### DISCUSSION THREAD

**Discussion by sudhiraroraCVS** [codefori/vscode-ibmi#1587] [Q&A]
Hi Team,

I work in CVS Health organization which has one of the biggest IBMi server and i am part of CI/CD discovery project with one of the objective is to evaluate VS code for different aspects of development and debug is crucial part of development, therefore we are exploring that as well on VS Code. We are currently not able to do debug on VS code yet.
As per documentation, there are below pre-requisites for debug to work :
1. Install Debug extension
2. Ensure all PTF’s installed
3. Generate Debug certificates
4. Start up the service.

We have completed Step1, Step2 and we are stuck on Step3 as we don’t have authority to create certificates. Our system team have authority but they don’t have VS code installed on their machine so they will not be able to do that.

We noticed as part of Generate step below 2 files are created at IFS location : /QIBM/ProdData/IBMiDebugService/bin/certs.

debug_service.pfx : is used to start up the service on the server.
debug_service.crt : which is used as a client certifcate. This can be downloaded and imported later.

By any chance can you provide us a downloadable files for these, so that we can share that with our system team to just download those files and upload on system at desired IFS location.

Please suggest if you have any other workaround for this situation.

Thanks
Sudhir.

**Comment by worksofliam** [comment]
Hi @sudhiraroraCVS 

Those files are unique to each server that you debug on. You should have a user with enough authority to create those certificates once, and then the users can download them through VS Code to their local device.

Find the debug settings, where you can import the file, in your connection settings.

<img width="1136" alt="image" src="https://github.com/codefori/vscode-ibmi/assets/3708366/92b65543-2ee8-405c-b059-e5462a92430c">

**Comment by sudhiraroraCVS** [comment]
Hi Barry,

I got the attached email to execute some shell commands. We have forwarded the same to our system tean to act on it and waiting for the response from them.



Sudhir Arora, ICP | CSM | Sr. Advisor, Encounter Management System(EMS), CVS/Health
p 480-707-6690 | c 469-406-2193
CVS Health | 10001 N 92nd St, Scottsdale, AZ 85258
[CSM Logo] ***@***.***

From: barry ***@***.***>
Sent: Friday, October 13, 2023 1:34 PM
To: codefori/vscode-ibmi ***@***.***>
Cc: Arora, Sudhir ***@***.***>; Mention ***@***.***>
Subject: [EXTERNAL] Re: [codefori/vscode-ibmi] Debug on VS Code for IBMi (Discussion #1587)

**** External Email - Use Caution ****
________________________________
Hi @sudhiraroraCVS<https://github.com/sudhiraroraCVS>

Those files are unique to each server that you debug on. You should have a user with enough authority to create those certificates once, and then the users can download them through VS Code to their local device.

Find the debug settings, where you can import the file, in your connection settings.
[image]<https://user-images.githubusercontent.com/3708366/275044522-92b65543-2ee8-405c-b059-e5462a92430c.png>
